first_hour_on_vwb/working_with_data.ipynb

```diff
@@ -0,0 +1,589 @@
+    "</div>\n",
+    "    \n",
+    "`workspace_setup.ipynb` creates two Cloud Storage buckets for your workspace files with workspace reference names:\n",
+    "\n",
+    "- ws_files\n",
+    "- ws_files_autodelete_after_two_weeks\n",
+    "\n",
+    "The code in this notebook will write output files to the \"autodelete\" bucket by default.  \n",
+    " Any file in this bucket will be automatically deleted <b>two weeks</b> after it is written.  \n",
+    " This alleviates the need for you to remember to clean up temporary and example files manually.  \n",
+    " If you want to write outputs to a durable location, simply change the assignment of the `BUCKET_REFERENCE` variable in the cell below and re-run the notebook.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Change this to \"ws_files\" to use the durable workspace bucket instead of the autodelete bucket.\n",
+    "BUCKET_REFERENCE = \"ws_files_autodelete_after_two_weeks\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "MY_BUCKET = get_bucket_url_from_reference(BUCKET_REFERENCE)\n",
+    "print(f'Bucket ID: {MY_BUCKET}')"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Working with data from referenced resources\n",
+    "\n",
+    "<a id='access-ref-data'></a>\n",
+    "\n",
+    "A [referenced resource](https://terra-docs.api.verily.com/docs/getting_started/web_ui/#referenced-vs-workspace-controlled-resources) points to a source outside of the current workspace in order to represent data or other elements in Verily Workbench. To add a referenced resource to your workspace, use the <a href=\"../working_with_resources.ipynb\">working_with_resources.ipynb</a> notebook.\n",
+    "\n",
+    "### Reading data from BigQuery\n",
+    "\n",
+    "<a id='add-bq-dataset'></a>\n",
+    "\n",
+    "There are many ways to interact with your referenced resource data in a Verily Workbench cloud environment. This notebook provides examples of several options, each of which are appropriate for certain use cases.\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Query with cell magic\n",
+    "\n",
+    "To quickly to run a query, consider using the [IPython magics for BigQuery](https://cloud.google.com/python/docs/reference/bigquery/latest/magics). Several examples are provided below. This option is useful for troubleshooting, demos and developing new queries, although it's not particularly useful for preserving and comparing longitudinal results of queries you anticipate running routinely.\n",
```

Comment:
```suggestion
    "To quickly to run a query, we provide several examples below using [IPython magics for BigQuery](https://cloud.google.com/python/docs/reference/bigquery/latest/magics). This option is useful for troubleshooting, demos and developing new queries.\n",
```
Just trying to get the necessary points across. If you feel strongly about preserving and comparing longitudinal results, we should point to examples that help with that.